supervisor/dispatchers.py

```diff
@@ -139,17 +137,23 @@ def _setup_logging(self, config, channel):
             warnings.warn("Specifying 'syslog' for filename is deprecated. "
                 "Use %s_syslog instead." % channel, DeprecationWarning)
             fmt = ' '.join((config.name, fmt))
-        self.mainlog = loggers.handle_file(
-            config.options.getLogger(),
-            filename=logfile,
-            fmt=fmt,
-            rotating=not not maxbytes, # optimization
-            maxbytes=maxbytes,
-            backups=backups)
+        if logfile is not None:
```

Comment:
This replaces the original comparison `if not logfile` which would evaluate to `False` on both `None`and an empty string.  This data originates from parsing the config file, so it an empty string may actually be possible (unsure).  Was this change intentional, and if so are you certain there is no code path where this may receive an empty string?

Reply:
The code now specially handles `logfile` being an empty string, in which case the behavior is the same as always.